From b6c5e3d94ca235c8af11dc9b4d62578739559071 Mon Sep 17 00:00:00 2001
From: Mingyen Hung <mingyen.hung@amlogic.com>
Date: Thu, 4 Oct 2018 18:53:20 +0800
Subject: [PATCH] tsync: expose tsync_video_started to user space

When slowsync_enable is 1, pcrscr will be reset
due to VIDEO_START event. This make pcrscr unstable.

Expose this value will make apps in userspace know
when pcrscr is stable to fetch.

Signed-off-by: Mingyen Hung <mingyen.hung@amlogic.com>
---
 drivers/amlogic/media/frame_sync/tsync.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/drivers/amlogic/media/frame_sync/tsync.c b/drivers/amlogic/media/frame_sync/tsync.c
index df091fe..c4c7537 100644
--- a/drivers/amlogic/media/frame_sync/tsync.c
+++ b/drivers/amlogic/media/frame_sync/tsync.c
@@ -1844,6 +1844,12 @@ static ssize_t show_firstvpts(struct class *class,
 	return sprintf(buf, "0x%x\n", timestamp_firstvpts_get());
 }

+static ssize_t show_tsync_video_started(struct class *class,
+		struct class_attribute *attr, char *buf)
+{
+	return sprintf(buf, "0x%x\n", tsync_video_started);
+}
+
 static ssize_t show_firstapts(struct class *class,
 		struct class_attribute *attr, char *buf)
 {
@@ -2019,6 +2025,8 @@ static struct class_attribute tsync_class_attrs[] = {
 	NULL),
 	__ATTR(checkin_firstapts, 0644, show_checkin_firstapts,
 	NULL),
+	__ATTR(tsync_video_started, 0444, show_tsync_video_started,
+	NULL),
 	__ATTR_NULL
 };

